test/built-ins/Temporal/PlainDateTime/prototype/with/options-empty.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2021 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindatetime.prototype.with
+description: Verify that undefined options are handled correctly.
+features: [Temporal]
+---*/
+
+const datetime = new Temporal.PlainDateTime(1976, 11, 18, 15, 23, 30, 123, 456, 789);
+const expected = "1976-11-05T15:23:30.123456789";
+
+assert.sameValue(
+  `${datetime.with({ day: 5 }, {})}`,
+  expected,
+  "options may be empty object"
+);
+
+assert.sameValue(
+  `${datetime.with({ day: 5 }, () => {})}`,
+  expected,
+  "options may be a function that returns an empty object"
```

Comment:
Note that "that returns an empty object" is not really relevant (and may be confusing). We're just trying to fetch properties off the function object, not calling the function.

Reply:
I've updated the test description -- how does it look now? (Thanks for clarifying -- I didn't realize we weren't calling the function.)